src/lib.rs

```diff
@@ -198,7 +202,14 @@ pub fn consumer(
                         }
 
                         let mut new_results = if gcov_type == GcovType::SingleFile {
-                            let new_results = try_parse!(parse_gcov(&gcov_path), work_item.name);
+                            let new_results = try_parse!(
+                                if gcov_ext.ends_with("gz") {
+                                    parse_gcov_gz(&gcov_path)
+                                } else {
```

Comment:
else endswith("gcov")
parse_gcov
else
log error

Reply:
the test is made on `gcov_ext` set to either "gcov.json.gz" or "gcov" so there is no other alternative.

Reply:
That's true now, but who knows how the code will evolve, the check would allow us to be more defensive.
It's a nit though, feel free to ignore ;)